bolt-sidecar/src/driver.rs

```diff
@@ -74,17 +96,24 @@ impl<C: StateFetcher, BLS: SignerBLS, ECDSA: SignerECDSA> SidecarDriver<C, BLS,
         let (api_events_tx, api_events_rx) = mpsc::channel(1024);
         CommitmentsApiServer::new(api_addr).run(api_events_tx).await;
 
+        // TODO: this can be replaced with ethereum_consensus::clock::from_system_time()
+        // but using beacon node events is easier to work on a custom devnet for now
+        // (as we don't need to specify genesis time and slot duration)
+        let head_tracker = HeadTracker::start(beacon_client.clone());
+
+        // Initialize the consensus clock.
+        let consensus_clock = from_system_time(
+            head_tracker.beacon_genesis_timestamp(),
+            cfg.chain.slot_time(),
+            SLOTS_PER_EPOCH,
+        );
```

Comment:
reminder: this is not great because head tracker initializes the genesis timestamp to 0.
We should add a fn to simply async fetch that value instead for safety

This can be done in another issue

Reply:
I added a comment for this.

but I don't understand how the soln for this should be.

```
let genesis_time = loop {
                    match beacon_client.get_genesis_details().await {
                        Ok(genesis_info) => break genesis_info.genesis_time,
                        Err(err) => {
                            warn!(?err, "failed to get genesis details");
                            sleep(RETRY_DELAY).await;
                            continue;
                        }
                    }
};
```

I am fetching the value like this

Reply:
we should really have this value in `ChainConfig` anyway - for now it's fine though 👍